### DISCUSSION TITLE
Best practices of project management for multiple users on Polyaxon v1

### DISCUSSION THREAD

**Discussion by shotarok** [polyaxon/polyaxon#1306] [General]
Hello all, I’m curious about the best practices of project management for multiple users on Polyaxon v1. Polyaxon v0 allows multiple users to have the same name project, but a project seems to be global under an organization on Polyaxon v1. The ideas I came up with are the following. Do we have other approaches?
* Add username as a prefix or suffix to a project name
* Use username as a tag in a project

**Comment by md-mq** [comment]
Let's make this `Discussion` a home for sharing and discussing patterns users are following to:

 * Organize and manage their projects 
 * Settings for CI/CD/CT 
 * Enable the rest of the community to share their thoughts on this important part of developing and scaling their workflows with Polyaxon
 * Enable auditing and access to protected data
 * Go from experimentation to deployment
 * Enable reproducibility
 * Enable collaboration
 
To set a template of how to share a pattern and to keep the `Discussion` organized:

 * Let's make each comment describe a pattern
 * Let's use threads to comment and ask questions about a specific pattern
 * Let's follow, loosely, the template below

> **Note**: Some of the patterns might be converted to tutorials or added as guides in the [scheduling strategies](https://polyaxon.com/docs/core/scheduling-strategies/) or [deployment strategies](https://polyaxon.com/docs/setup/deployment-strategies/) in the documentation.

Template

```
## Overview

> A description of the process or the pattern

## Local project structure

> Folders and files structure

## Writing/Authoring polayxonfiles

> YAML vs Python
> Using presets?
> Using `pathRef`, `UrlRef`, `HubRef`, or a built-in component
> Component vs Operation
> Forking public hub components?

## Development

> How do you submit files to the API/Scheduler during devlopement?
> Do you use git or upload?
> Do you require rebuilding images?
> Do you generally delete test files and test runs?
> How do you manage tags?
> Does Polyaxon need some extra tags management on the project/org level?
> Does Polyaxon need some special tags: like `prod`/`stage`/`testing` similar to the model and the component versions?
> Does Polyaxon need some special dropdown to allow users to easily switch the view to prod/test, and the space should be preserved throughout the browsing experience?

## Testing

> How do you test your files and the logic you are running?
> Do you declare a test data as a second connection and do you manually need to deploy a polyaxonfile with that subset/test data?
> Do you run the logic with NO_OP or OFFLINE?

## Deployment

> How do you handle polyaxonfiles discovery?
> Do you write descriptions and do you deploy with new descriptions for each run? 
> How do you handle building images?
> Do you submit polyaxonfiles with external systems?

## Presets

> Do you write override files?
> Do you save them globally or on each repo?
> Do you use presets to switch between test and prod?

## Agents or Multi-deployment clusters

> How do you manage multi-environments?
> How do you manage data access?
> How do you manage access to production connections?
> Do you mount specific configmaps/variables for production?

## Limitations

> Are their any limitations with this pattern?
> Do you feel like Polyaxon can expose some abstractions to simplify this pattern?
```
